stream/replicator.h

```diff
@@ -152,6 +156,42 @@ class SubscribableRemoteStream final {
       }
     }
 
+    template <SubscriptionMode MODE>
```

Comment:
IIRC, you could write `template <SubscriptionMode MODE = SM>` and avoid specifying `<SM>` down below when you call it.